sound/soc/sof/intel/hda-loader.c

```diff
@@ -372,3 +372,106 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	dev_err(sdev->dev, "error: load fw failed err: %d\n", ret);
 	return ret;
 }
+
+static int cl_dsp_init_skl(struct snd_sof_dev *sdev)
+{
+	const struct sof_intel_dsp_desc *chip = sdev->hda->desc;
+	int ret;
+
+	/* check if the core is already enabled, if yes, reset and make it run,
+	 * if not, powerdown and enable it again.
+	 */
+	if (hda_dsp_core_is_enabled(sdev, HDA_DSP_CORE_MASK(0))) {
+		/* if enabled, reset it, and run the core. */
+		ret = hda_dsp_core_stall_reset(sdev, HDA_DSP_CORE_MASK(0));
+		if (ret < 0)
+			goto err;
+
+		ret = hda_dsp_core_run(sdev, HDA_DSP_CORE_MASK(0));
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: dsp core start failed %d\n",
+				ret);
+			ret = -EIO;
```

Comment:
No need to change the value of ret here, otherwise the meaning will be lost further up the stack. So best remove this line.